aten/src/ATen/native/sparse/SparseCsrTensorMath.cpp

```diff
@@ -237,6 +237,21 @@ CREATE_UNARY_UFUNC_NO_INPLACE(signbit);
 CREATE_UNARY_UFUNC_FUNCTIONAL(isnan);
 CREATE_UNARY_UFUNC_FUNCTIONAL(isinf);
 
+// pow function for Sparse CSR, since square is composite - this will also enable square for Sparse CSR
+Tensor& pow_sparse_csr_scalar_out(const Tensor& self, const Scalar& exponent, Tensor& result) {
+  return unary_op_out<Tensor& (*)(const Tensor&, const Scalar&, Tensor&)>(&at::pow_outf, self, result, exponent);
+}
+
+Tensor pow_sparse_csr_scalar(const Tensor& self, const Scalar& exponent) {
+  return get_result_tensor_for_unary_op<Tensor (*)(const Tensor&, const Scalar&)>(&at::pow, self, exponent);
```

Comment:
`pow` only works for exponents `> 0`, right? Otherwise the rule that 0 maps to 0 is broken.

Reply:
Great catch! Let's restrict these functions to preserve zeros.

Reply:
> Looks like square will have to be implemented separately, and pow will have to be left alone for now.

Why is that? We can raise an error when `exponent == 0` and do the computation for all other values.

Reply:
Yep, right. Thanks, @IvanYashchuk